app/Http/Controllers/Api/ApiV1Controller.php

```diff
@@ -4494,4 +4494,95 @@ public function statusUnpin(Request $request, $id)
 
         return $this->json($status);
     }
+
+    /**
+     * GET /api/v1/notifications
+     *
+     * @return array
+     */
+    public function accountNotificationsMarkAsRead(Request $request)
+    {
+        /* abort_if(! $request->user(), 403);
+        abort_unless($request->user()->tokenCan('write'), 403);*/
+
+        $user = $request->user();
+        $pid = $user->profile_id;
+        $id = $request->input('id');
+
+        if (empty($id)) {
+            $notifications = Notification::whereNull('read_at')
+                ->whereProfileId($pid)
+                ->get();
+
+            foreach ($notifications as $n) {
+                $n->read_at = now();
+                $n->save();
+                Cache::forget('service:notification:'.$n->id);
+            }
+
+            return $this->json(['success' => true]);
+        }
+
+        $notification = Notification::whereProfileId($pid)->findOrFail($id);
+        $notification->read_at = 'now()';
+        $notification->save();
+
+        Cache::forget('service:notification:'.$id);
+        return $this->json(['success' => true]);
+    }
+
+    /**
+     * POST /api/v1/notifications/mark_as_unread
```

Comment:
What is the point of this? I don't think we need to support marking a notification as unread. Please remove this.